src/CommunityToolkit/Xamarin.CommunityToolkit/Effects/Shadow/PlatformShadowEffect.uwp.cs

```diff
@@ -0,0 +1,201 @@
+﻿using System;
+using System.ComponentModel;
+using System.Linq;
+using System.Numerics;
+using Windows.UI.Composition;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls;
+using Windows.UI.Xaml.Hosting;
+using Windows.UI.Xaml.Shapes;
+using Xamarin.CommunityToolkit.Effects;
+using Xamarin.CommunityToolkit.UWP.Effects;
+using Xamarin.Forms;
+using Xamarin.Forms.Platform.UWP;
+using Grid = Xamarin.Forms.Grid;
+using Image = Windows.UI.Xaml.Controls.Image;
+
+[assembly: ExportEffect(typeof(PlatformShadowEffect), nameof(ShadowEffect))]
+
+namespace Xamarin.CommunityToolkit.UWP.Effects
+{
+	public class PlatformShadowEffect : PlatformEffect
+	{
+		enum ShadowEffectState
+		{
+			Initialized,
+			PanelCreated,
+			Attached
+		}
+
+		const float defaultRadius = 10f;
+
+		const float defaultOpacity = 1f;
+
+		ShadowEffectState state;
+
+		SpriteVisual? spriteVisual;
+
+		Layout<View>? shadowPanel;
+
+		DropShadow? shadow;
+
+		FrameworkElement View => Control ?? Container;
+
+		protected override void OnElementPropertyChanged(PropertyChangedEventArgs args)
+		{
+			base.OnElementPropertyChanged(args);
+
+			switch (args.PropertyName)
+			{
+				case nameof(ShadowEffect.ColorPropertyName):
+				case nameof(ShadowEffect.OpacityPropertyName):
+				case nameof(ShadowEffect.RadiusPropertyName):
+				case nameof(ShadowEffect.OffsetXPropertyName):
+				case nameof(ShadowEffect.OffsetYPropertyName):
+					UpdateShadow();
+					break;
+			}
+		}
+
+		protected override void OnAttached()
+		{
+			if (!(Element is View elementView))
+				return;
+
+			switch (state)
+			{
+				case ShadowEffectState.Attached:
+					return;
+
+				case ShadowEffectState.Initialized:
+					shadowPanel = new StackLayout()
+					{
+						Children = { new Grid() }
+					};
+					state = ShadowEffectState.PanelCreated;
+
+					MoveElementTo(elementView, shadowPanel);
+					return;
+
+				case ShadowEffectState.PanelCreated:
+					AppendShadow();
+					state = ShadowEffectState.Attached;
+					return;
+
+				default:
+					throw new ArgumentOutOfRangeException();
+			}
+		}
+
+		protected override void OnDetached()
+		{
+			if (state != ShadowEffectState.Attached)
+				return;
+
+			View.SizeChanged -= ViewSizeChanged;
+			shadow?.Dispose();
+			shadow = null;
+			spriteVisual?.Dispose();
+			spriteVisual = null;
+
+			state = ShadowEffectState.PanelCreated;
+		}
+
+		void AppendShadow()
+		{
+			var view = ElementCompositionPreview.GetElementVisual(View);
+
+			var compositor = view.Compositor;
+			shadow ??= compositor.CreateDropShadow();
+			UpdateShadow();
+			spriteVisual = compositor.CreateSpriteVisual();
+			spriteVisual.Shadow = shadow;
+			spriteVisual.Size = View.ActualSize;
+			spriteVisual.Size = new Vector2(1170, 40);
+
+			View.SizeChanged += ViewSizeChanged;
+
+			var renderer = shadowPanel?.Children.First().GetOrCreateRenderer();
```

Comment:
Are we ok to throw an exception if there are no Children?

Reply:
This can't happen. Because I initialized shadowPanel with one child (Grid) always